pkgs/games/starruler2/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  callPackage,
+  makeWrapper,
+  # Path to directory containing music for the game,
+  # such as `data/music` from the commercial version.
+  musicPath ? null,
+  # Paths of mods to add.
+  modPaths ? [],
+}: let
+  data = callPackage ./data.nix {};
+  engine = callPackage ./engine.nix {};
+in
+  # The game cannot access files outside the working directory,
+  # so it cannot access files behind symlinks.
+  stdenv.mkDerivation {
+    pname = "starruler2";
+    version = lib.trivial.max data.version engine.version;
+
+    phases = ["installPhase"];
+
+    nativeBuildInputs = [makeWrapper];
+
+    installPhase = ''
+      game="$out/share/games/starruler2"
+      mkdir -p "$out/bin"
+
+      cp -r "${data}/share" "$out/share"
+      chmod -R u+w "$out/share"
+
+      ${
+        if musicPath != null
+        then ''
+          cp -r "${musicPath}" "$game/data/music"
+        ''
+        else ""
+      }
+
+      ${
+        if modPaths != []
+        then ''
+          cp -r ${lib.concatStringsSep " " (map (path: ''"${path}"'') modPaths)} "$game/mods"
+        ''
+        else ""
+      }
+
+      makeWrapper "${engine}/bin/starruler2" "$out/bin/starruler2" --chdir "$game"
+    '';
+
+    meta = with lib; {
+      description = "A massive-scale 4X/RTS game set in space";
+      homepage = "https://github.com/OpenSRProject";
+      license = with licenses; [mit cc-by-nc-20]; # Art assets are licensed as CC-BY-NC 2.0.
```

Comment:
If you're so set on having here the licenses of the engine and the data, it would be better to have this:
```
license = lib.toList engine.licenses ++ lib.toList data.licenses;
```

Reply:
Then MIT would be listed twice. As far as I am aware, the `license` list is not deduplicated.

Reply:
`lib.intersectLists`

Reply:
Wouldn't an union make more sense though ? I don't see any function like `lib.unionLists`, but `lib.unique` could be used

Reply:
ah sorry, yes you're right

Reply:
Done using `lib.unique`.